packages/itwin/grouping-mapping-widget/src/widget/components/ExtractionMessageModal.tsx

```diff
@@ -0,0 +1,125 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { Button, DefaultCell, Icon, Modal, ModalButtonBar, ModalContent, Table, tableFilters, Text } from "@itwin/itwinui-react";
+import React, { useEffect, useMemo, useState } from "react";
+import type { CreateTypeFromInterface } from "../utils";
+import type { ExtractionMessageData } from "./hooks/useMappingsOperations";
+import "./ExtractionMessageModal.scss";
+import { SvgClock } from "@itwin/itwinui-icons-react";
+import type { CellProps, CellRendererProps, Column } from "react-table";
+import { StatusIcon } from "./StatusIcon";
+import { ExtractionLogCustomFilter } from "./ExtractionLogCustomFilter";
+
+export interface ExtractionMessageModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  extractionMessageData: ExtractionMessageData[];
+  timestamp: string;
+}
+
+export const ExtractionMessageModal = ({ isOpen, onClose, extractionMessageData, timestamp }: ExtractionMessageModalProps) => {
+  const [formattedTimestamp, setFormattedTimestamp] = useState<string>("");
+  useEffect(() => {
+    formatTimestamp;
```

Comment:
What does this do?

Reply:
The hook will call the function where the stamp is formatted as soon as it retrieves the timestamp at the moment the component mounts.

Reply:
The hook is not doing anything. `formatTimestamp` is not being invoked.

Reply:
I also think it would be easier to read if that logic was instead in the `useEffect` itself rather then in that `useMemo`.

Reply:
Oh, you're right, since it's in a `useMemo`.

Reply:
I have moved it inside `useEffect`.